run-tests.php

```diff
@@ -2781,7 +2781,9 @@ function run_test(string $php, $file, array $env): string
             $orig_shortname = str_replace(TEST_PHP_SRCDIR . '/', '', $file);
             $diff = "# original source file: $orig_shortname\n" . $diff;
         }
-        show_file_block('diff', $diff);
+        if (!$SHOW_ONLY_GROUPS || array_intersect($restype, $SHOW_ONLY_GROUPS)) {
```

Comment:
Thank you @iluuu1994 for taking #8392 over.

- this/run-tests.php change is some rebase leftover
- to discuss/merge test changes, please use #9744
- do not forget to add nightly job one this PR is working
- please keep https://github.com/mvorisek/php-src/blob/fea5e38b3cadcd43199f3f5733308425cab606a1/.github/scripts/windows/test_task.bat#L3-L6 to prevent the CI .bat files to be run locally (there can mutate some system configuration when run locally by a mistake)
- `actions/cache`, ex. https://github.com/atk4/core/blob/develop/.github/workflows/test-unit.yml#L34-L45, should be used to speed the CI

Reply:
> this/run-tests.php change is some rebase leftover

This is intentional, otherwise the `-g` flag doesn't behave properly (diff gets printed even if excluded). It's backported from master.

> do not forget to add nightly job one this PR is working

It's removed intentionally for now until this has proven to work fine, I'll re-add it after.